lib/Doctrine/Common/Lexer/AbstractLexer.php

```diff
@@ -160,7 +162,7 @@ public function isNextToken($token)
     /**
      * Checks whether any of the given tokens matches the current lookahead.
      *
-     * @param string[] $tokens
+     * @param list<int|string|null> $tokens
```

Comment:
I don't know if it makes sense to allow `null` here, but I'm forced to do so if a token type can be passed here. Either this is correct or there should be a subtype of `Token` that disallows `null`. Maybe we should also rename `$tokens` to `$tokenTypes` to make things less confusing.

Reply:
Looking a bit more into the code, I think the type will be null if it is not understood by the lexer (no condition in `getType` matches the value). I don't think it makes sense to check if that's the case, so let me create another type just for that.

Reply:
> Maybe we should also rename $tokens to $tokenTypes to make things less confusing.

`skipUntil` uses `$type`, so let's be consistent with that.